azurerm/resource_arm_azure_firewall_network_rule_collection.go

```diff
@@ -0,0 +1,342 @@
+									string(network.TCP),
+									string(network.UDP),
+								}, true),
+							},
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							Set:              schema.HashString,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmAzureFirewallNetworkRuleCollectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).azureFirewallsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	firewallName := d.Get("azure_firewall_name").(string)
+
+	azureRMLockByName(firewallName, azureFirewallResourceName)
+	defer azureRMUnlockByName(firewallName, azureFirewallResourceName)
+
+	firewall, err := client.Get(ctx, resourceGroup, firewallName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	ipConfigs := fixArmAzureFirewallIPConfiguration(&firewall)
+	firewall.AzureFirewallPropertiesFormat.IPConfigurations = &ipConfigs
+
+	newFwRuleCol := expandArmAzureFirewallNetworkRuleCollection(d)
+	ruleCollections := append(*firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections, newFwRuleCol)
+	existingCollection, index, exists := findArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	if exists {
+		if name == *existingCollection.Name {
+			ruleCollections = append(ruleCollections[:index], ruleCollections[index+1:]...)
+		}
+	}
+	firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections = &ruleCollections
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, firewallName, firewall)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for creation/update of Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	read, err := client.Get(ctx, resourceGroup, firewallName)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read Azure Firewall %q (Resource Group %q) ID", firewallName, resourceGroup)
+	}
+
+	var collectionID string
+	for _, collection := range *read.AzureFirewallPropertiesFormat.NetworkRuleCollections {
+		if *collection.Name == name {
+			collectionID = *collection.ID
+		}
+	}
+	d.SetId(collectionID)
+
+	return resourceArmAzureFirewallNetworkRuleCollectionRead(d, meta)
+}
+
+func resourceArmAzureFirewallNetworkRuleCollectionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).azureFirewallsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	name := d.Get("name").(string)
+	fwName := d.Get("azure_firewall_name").(string)
```

Comment:
we need to parse these values from the ID here, so once the change to the ID is made above, can we make this:

```
id, err := parseAzureResourceID(d.Id())
if err != nil {
  return err
}

name := id.Path["networkRuleCollection"]
firewallName := id.Path["azureFirewalls"]
resourceGroup := id.ResourceGroup
```